vsintegration/tests/FSharp.Editor.Tests/Hints/HintTestFramework.fs

```diff
@@ -57,9 +62,19 @@ module HintTestFramework =
     let getHints (document: Document) hintKinds =
         async {
             let! ct = Async.CancellationToken
+
+            let getTooltip hint =
+                async {
+                    let! roslynTexts = hint.GetTooltip document ct
+                    return roslynTexts |> Seq.map (fun roslynText -> roslynText.Text) |> String.concat ""
+                }
+
             let! sourceText = document.GetTextAsync ct |> Async.AwaitTask
             let! hints = HintService.getHintsForDocument sourceText document hintKinds "test" ct
-            return hints |> Seq.map convert
+
+            return
+                hints
+                |> Seq.map (fun hint -> hint |> (getTooltip >> Async.RunSynchronously) |> convert hint)
         }
```

Comment:
@vzarytovskii please take a look here, I feel I am complicating the matters here with all the async stuff

Reply:
Yes, please don't use `RunSynchronously` within `async`. 

You can do something like:

```fsharp
let! tooltips = hints |> Seq.map getTooltip |> Async.Parallel
return (hints, tooltips) ||> Seq.zip |> Seq.map convert // <- change convert to accept tuple
```

Reply:
Okay that's done I think